src/CollisionDetection.hs

```diff
@@ -1,103 +1,98 @@
 module CollisionDetection
   ( wallsCollision
   , bricksCollision
+  , paddleCollision
   , CollisionSide (..)
   ) where
 
 
-  import GameBoard
+import GameBoard
 
+-- | If there a collision on which side it accured
+data CollisionSide =
+      TopSide | BottomSide | LeftSide | RightSide
+      deriving Show
 
-  -- | If there a collision on which side it accured
-  data CollisionSide =
-    TopSide | BottomSide | LeftSide | RightSide
-    deriving Show
+-- | Given position and radius of the ball, return nothing if there is
+--   no collision or the collisionSide.
```

Comment:
You can use 'CollisionSide' in Haddock comments to automatically get hyperlinks to the relevant type in the documentation. (Use 'Fully.Qualified.Name' if the type is not in scope.)